amethyst_network/src/server/server.rs

```diff
@@ -0,0 +1,105 @@
+use super::{ReceiveHandler, SendHandler, ServerConfig, ServerSocketEvent, UdpReceiver, UdpSender};
+use laminar::{error::NetworkResult, DeliveryMethod, Packet};
+use std::net::SocketAddr;
+use std::sync::Arc;
+
+/// 'Server' abstracts TCP and UDP sockets away.
+pub struct Server {
+    config: ServerConfig,
+    // Handler to access the internals of the UDP receiving thread
+    udp_receiver: Option<ReceiveHandler>,
+    // Handler to access the internals of the TCP receiving thread
+    tcp_receiver: Option<ReceiveHandler>,
+    // Handler to access the internals of the UDP sender thread
+    udp_sender: Option<SendHandler>,
+    // Handler to access the internals of the TCP sender thread
+    tcp_sender: Option<Arc<SendHandler>>,
+}
+
+impl Server {
+    /// 'Server' abstracts TCP and UDP sockets away into an uniformed API.
+    pub fn new(config: ServerConfig) -> Self {
+        Server {
+            config: config.clone(),
+
+            udp_receiver: None,
+            tcp_receiver: None,
+
+            udp_sender: None,
+            tcp_sender: None,
+        }
+    }
+
+    /// This will start the server.
+    /// 1: Fire up a TCP-sender and TCP-receiver if enabled in config.
+    /// 2: Fire up a UDP-sender and UDP-receiver.
+    /// 3: Set up some `channels` to communicate with underlying threads.
+    ///
+    /// The method uses the config provided when creating a `Server` instance.
+    pub fn run(&mut self) {
```

Comment:
from what I noticed above we could probably just merge run with new? This seems like an OOP style which is not really needed.